### PR TITLE
Add Kernel Descriptions and Example Syntax

### PR DESCRIPTION
**Pull request by lindsayad**
Adds Descriptions and Example Syntax documentation for all of the framework kernels.

Also:

- Move `mass_lumping` tests from `modules/misc` into the framework test suite. Seems appropriate since `MassLumpingTimeDerivative` is a framework class
- Move `Phase Field Equations.md` to `Phase_Field_Equations.md` so links work

### CODE REVIEW

File: docs/content/documentation/systems/Kernels/framework/AnisotropicDiffusion.md [hunk 1]
```diff
@@ -1,7 +1,37 @@
 <!-- MOOSE Documentation Stub: Remove this when content is added. -->
```
Comment (dschwen) [inline docs/content/documentation/systems/Kernels/framework/AnisotropicDiffusion.md]: You... should probably remove all those

File: docs/content/documentation/systems/Kernels/framework/BodyForce.md [hunk 2]
```diff
@@ -1,6 +1,35 @@
...
+ Since `value` and `function` were not supplied, they default to 1 and 1. The
+ corresponding `Postprocessor` block is
+
+ !listing test/tests/kernels/2d_diffusion/2d_diffusion_bodyforce_test.i
+ block=Postprocessors label=false
+
+ Note that in this test the postprocessor value is actually calculated from a
+ function.
+
 !syntax description /Kernels/BodyForce
```
Comment (dschwen) [inline docs/content/documentation/systems/Kernels/framework/BodyForce.md]: Shouldn't the class description come first (before the manual prose)?

File: docs/content/documentation/systems/Kernels/framework/CoupledTimeDerivative.md [hunk 3]
```diff
@@ -1,7 +1,31 @@
 <!-- MOOSE Documentation Stub: Remove this when content is added. -->
 
 # CoupledTimeDerivative
-!syntax description /Kernels/CoupledTimeDerivative
```
Comment (dschwen) [inline docs/content/documentation/systems/Kernels/framework/CoupledTimeDerivative.md]: Why remove those?

File: docs/content/documentation/systems/Kernels/framework/MaterialDerivativeTestKernel.md [hunk 4]
```diff
@@ -1,7 +1,49 @@
 <!-- MOOSE Documentation Stub: Remove this when content is added. -->
 
 # MaterialDerivativeTestKernel
-!syntax description /Kernels/MaterialDerivativeTestKernel
+
+## Description
+
+`MaterialDerivativeTestKernel` is mostly useful as a demonstration object. Its
```
Comment (dschwen) [inline docs/content/documentation/systems/Kernels/framework/MaterialDerivativeTestKernel.md]: No, it is a generic test Kernel, that allows checking the correctness of the derivative material properties using the built in Jacobian checks in the test harness!

File: test/src/kernels/FuncCoefDiffusion.C [hunk 5]
```diff
@@ -30,13 +30,13 @@ FuncCoefDiffusion::FuncCoefDiffusion(const InputParameters & parameters)
 Real
 FuncCoefDiffusion::computeQpResidual()
 {
-  Real k = _function.value(_t, _qp);
```
Comment (dschwen) [inline test/src/kernels/FuncCoefDiffusion.C]: Holy dooley

Comment (permcody) [reply test/src/kernels/FuncCoefDiffusion.C]: image

File: docs/content/documentation/systems/Kernels/framework/CoupledTimeDerivative.md (hunk 3 above)
Comment (lindsayad) [reply docs/content/documentation/systems/Kernels/framework/CoupledTimeDerivative.md]: Well, you could argue that I needlessly duplicated my effort but the class description and the information I filed under `## Description` are generally redundant. If we're fine with the redundancy, I'll put them back in.

File: docs/content/documentation/systems/Kernels/framework/BodyForce.md [hunk 6]
```diff
@@ -1,7 +1,32 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
-
 # BodyForce
-!syntax description /Kernels/BodyForce
+
+## Description
+
+`BodyForce` implements a force term in momentum transport or structural
+mechanics or a source term in species/mass transport. Its weak form looks like
+$$(\psi_i, -f)$$ where $f$ is the magnitude of the force or source. The force is
```
Comment (aeslaughter) [inline docs/content/documentation/systems/Kernels/framework/BodyForce.md]: Can you update the $f$ to also discuss that this multiples value * pp * function.

**Review by aeslaughter — DISMISSED** [review]
These minor changes with what I gave you in a PR would be great. Thanks for doing this!

**Comment by lindsayad** [comment]
@aeslaughter I think this could be ready to go.

**Review by aeslaughter — APPROVED** [review]

**Comment by sapitts** [comment]
@lindsayad and @aeslaughter this commit introduces links to `KKSSplitChCRes.md` and `KKS Derivations.md` which are causing failures in the Bison documentation system build.  Is the current solution to add the path to these two files to the bison `doc/content.yml` file as @aeslaughter did for the fracture integral stuff in Mastodon yesterday?

**Pull request merged**